packages/components/src/popover/index.js

```diff
@@ -407,22 +428,25 @@ const Popover = (
 			<div className="components-popover__content">{ children }</div>
 			{ hasArrow && (
 				<div
-					className="components-popover__arrow"
 					ref={ arrowRef }
+					className={ [
+						'components-popover__arrow',
+						`components-popover__arrow--${
+							placementData.split( '-' )[ 0 ]
+						}`,
+					].join( ' ' ) }
 					style={ {
-						left:
-							! arrowData?.x || Number.isNaN( arrowData?.x )
-								? 0
-								: arrowData.x,
-						top:
-							! arrowData?.y || Number.isNaN( arrowData?.y )
-								? 0
-								: arrowData.y,
-						right: undefined,
-						bottom: undefined,
-						[ staticSide ]: '-4px',
+						left: Number.isFinite( arrowData?.x )
+							? `${ arrowData.x }px`
+							: '',
+						top: Number.isFinite( arrowData?.y )
+							? `${ arrowData.y }px`
+							: '',
+						'--wp-popover-arrow-size': '14px',
```

Comment:
So `14px` is our magic number and I guess it's the one to match our previous design, correct? 

Since this number is not affected by anything and is not configurable, is there a reason to have it here and not in the scss file directly?

Reply:
> So `14px` is our magic number and I guess it's the one to match our previous design, correct?

Short answer, yes!

Long answer: after a few trials, it looks like, in order to match our previous designs:

- in terms of arrow proportions, the height of the triangle needs to be roughly half of its base (which reflects in the path coordinates in the SVG)
- the previous "arrow size" before the popover refactor was `8px`
- therefore, I factored a height for the triangle of `7px`(plus `1px` border)
- therefore, the base of the triangle is 7 * 2 = `14px`

> Since this number is not affected by anything and is not configurable, is there a reason to have it here and not in the scss file directly?

Good question! When I started working on this PR, most of the arrow styles where applied inline and required some form of math in JS (mostly to get the triangle height, knowing the target triangle base). And so, I decided to use a CSS custom prop to share the value between JS and SCSS.

But now that you point it out, there isn't really a need for it — so I switched to a SCSS variable (which follows more the style of the component) in https://github.com/WordPress/gutenberg/pull/42874/commits/de33f60b3ce48ebcc9edc53bab00f02cab86db7a